docs/new_predicate_guidelines.md

```diff
@@ -74,3 +75,4 @@ Our vetting process is simple.
 [predicateType]: ../spec/v1/predicate.md#fields
 [protobuf definition]: ../protos/README.md
 [vetting process]: #vetting-process
+[predicate template]: /spec/predicates/template/template.md
```

Comment:
Nit: Do we expect other templates in the future? I'm curious about the choice to place the template file in a sub-directory.

Reply:
I don't expect more templates, I just wanted to separate it from the other predicate definitions. I also wonder if we ever foresee a template predicate but I suspect not. Should I move it up a directory?

Reply:
Yes, in fact, since it's in the same directory as the predicates, it might be a bit confusing as currently stored. In fact, I'm comfortable having this become `/spec/v1/predicate_template.md` since this may change in future versions. Wdyt?

Reply:
I'm not sure if this belongs in the v1 directory. The new predicate guidelines aren't part of the spec, they're in docs/ and predicates are also independently versioned. I could just rename this as _TEMPLATE.md or something and put it in the same directory as all the other predicate specs.

Reply:
I tend to think of the predicate template as part of the actual spec because all predicates defined under v1 follow the same format, regardless of their version, whereas the new predicate guidelines are in docs/ because those are more fluid guidelines. In other words, I might expect the predicate format to evolve between different versions of the spec, while the guidelines shouldn't have an impact on the format of predicates. That said, I don't want to block this PR on this particular point either; we can always move files around the repo later, if need be.